src/Adapters/NLogTarget.Shared/ApplicationInsightsTarget.cs

```diff
@@ -153,11 +153,21 @@ private void BuildPropertyBag(LogEventInfo logEvent, ITelemetry trace)
                 {
                     string key = keyValuePair.Key.ToString();
                     object valueObj = keyValuePair.Value;
-                    if (valueObj != null)
+                    if (valueObj == null)
                     {
-                        string value = valueObj.ToString();
-                        propertyBag.Add(key, value);
+                        continue;
                     }
+
+                    string value = valueObj.ToString();
+                    if (propertyBag.ContainsKey(key))
+                    {
+                        if (value == propertyBag[key])
+                        {
+                            continue;
+                        }
+                        key += "_1";
```

Comment:
Glad to see the diff resolved. :)
I think this would be safer if it handled _N instead of just _1.
Since the scenario of an existing key should not be the normal path I wouldn't worry about optimizing for a single duplicate hardcoded as _1 but would instead either keep a counter while looking for an unused key or just take the current number of clock ticks or something fast and unique to use as a suffix if you are concerned you will have a lot of duplicates.
The interesting question will be if you really only have one duplicate, and if so why, or if you can actually run into a situation where there are many duplicates in which case it seems there is something else to investigate as to why that is happening.